### PR TITLE
RAC-329: Prepare workshop

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

- Fixing CONTRIBUTING.md
- Moving forwardRef & ...rest props to Component's unit tests and removing the automated one (RAC-332)
- Fixing Badge whitespace

Before:
image
After:
image

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: .github/workflows/dsm-test.yml [hunk 1]
```diff
@@ -86,6 +86,7 @@ jobs:
         run: cd akeneo-design-system/example && yarn install
 
       - name: Launch example integration tests
+        id: example_integration_tests
         run: cd akeneo-design-system/example && yarn test
```
Comment (StevenVAIDIE) [inline .github/workflows/dsm-test.yml:90]: I not sure about the added line on visual_test_execution but should we add :
```suggestion
        run: cd akeneo-design-system/example && yarn test && echo ::set-output name=status::success
```

File: .github/workflows/dsm-test.yml [hunk 2]
```diff
@@ -98,27 +99,26 @@ jobs:
       # In this next section (only launched if the visual regression failed) we will update the visual snapshots
       # to provide a pull request to the developer contributing to the DSM.
       - name: Remove files generated by previous steps
-        if: failure() && steps.visual_test_execution.outcome == 'failure'
+        if: failure() && steps.visual_test_execution.outcome == 'failure' && steps.example_integration_tests.outcome == 'success'
```
Comment (StevenVAIDIE) [inline .github/workflows/dsm-test.yml:102]:
```suggestion
        if: failure() && steps.visual_test_execution.outputs.status == 'failure' && steps.example_integration_tests.outputs.status == 'success'
```

This is the objective to add 
`echo ::set-output name=status::success`
no ?

File: akeneo-design-system/CONTRIBUTING.md [hunk 3]
```diff
@@ -6,42 +6,52 @@ As this design system will be used among multiple products and multiple teams, w
...
-- Write the related Story of your Component `src/components/ComponentName/ComponentName.stories.tsx`
-  - Get the template from [Story template](#story-template) and replace `ComponentName` by your Component name
-  - Write `Usage introduction` and `General guidance` paragraph
+  - Make sure `forwardRef` management works properly on your Component ([https://reactjs.org/docs/forwarding-refs.html](https://reactjs.org/docs/forwarding-refs.html)) (should be automatically done by the generator)
+  - Make sure you forward ...props to your Component (should be automatically done by the generator)
+  - Make your Component accessible ([https://developer.mozilla.org/en-US/docs/Web/Accessibility](https://developer.mozilla.org/en-US/docs/Web/Accessibility))
+    - Your component should react to keyboard events if relevant
+    - Your component should be describable (`aria-label`, `alt`, etc) and readable by screen readers
+    - Your component should be focusable if relevant.
+- Write the related Story of your Component `src/components/ComponentName/ComponentName.stories.tsx` (should be automatically ganarated by the generator)
```
Comment (StevenVAIDIE) [inline akeneo-design-system/CONTRIBUTING.md:32]:
```suggestion
- Write the related Story of your Component `src/components/ComponentName/ComponentName.stories.tsx` (should be automatically generated by the generator)
```

File: akeneo-design-system/generator/components/MyComponent/MyComponent.unit.tsx [hunk 4]
```diff
@@ -8,3 +8,17 @@ test('it renders its children properly', () => {
 
   expect(getByText('MyComponent content')).toBeInTheDocument();
 });
+
+// Those tests should pass directly if you follow the contributing guide.
+// If they fail, maybe you need to customize your implementation forward ref and ...rest props
```
Comment (ValentinMumble) [inline akeneo-design-system/generator/components/MyComponent/MyComponent.unit.tsx:13]:
```suggestion
// If you add required props to your Component, these tests will fail and you will need to add these required props here as well
```

**Review by StevenVAIDIE — APPROVED** [review]

File: akeneo-design-system/src/components/Button/Button.unit.tsx [hunk 5]
```diff
@@ -55,3 +55,33 @@ it('it does not call onClick handler when user hit enter key on button', () => {
...
 });
+
+describe('Button supports forwardRef', () => {
+  const ref = {current: null};
+
+  render(
+    <Button
+      onClick={() => {
+        // Do nothing
+      }}
```
Comment (ValentinMumble) [inline akeneo-design-system/src/components/Button/Button.unit.tsx:66]:
```suggestion
      onClick={jest.fn()}
```

**Review by ValentinMumble — APPROVED** [review]

File: .github/workflows/dsm-test.yml (hunk 1 above)
Comment (ValentinMumble) [reply .github/workflows/dsm-test.yml]: yes

File: .github/workflows/dsm-test.yml (hunk 2 above)
Comment (ValentinMumble) [reply .github/workflows/dsm-test.yml]: yes

**Pull request merged**